src/lib/marlin_plonk_bindings/stubs/src/bigint_256.rs

```diff
@@ -126,8 +126,10 @@ pub fn caml_bigint_256_of_bytes(x: &[u8]) -> Result<BigInteger256, ocaml::Error>
     if x.len() != len {
         ocaml::Error::failwith("caml_bigint_256_of_bytes")?;
     };
-    let x = BigInteger256::deserialize(&mut &x[..]).expect("deserialize failed");
-    Ok(x)
+    match BigInteger256::deserialize(&mut &x[..]) {
+        Ok(b) => Ok(b),
+        Err(_) => Err(ocaml::Error::invalid_argument("caml_bigint_256_of_bytes, deserialize failed").err().unwrap())
```

Comment:
I don't think the invalid_argument should be used like this, as it returns a `()` as result. I think you might just want to write this:

```rust
Err(_) => ocaml::Error::Message("deserialization error")
```

Reply:
note that you also rewrite that block as:

```rust
BigInteger256::deserialize(&mut &x[..]).map_err(|_| ocaml::Error::Message("deserialization error"))
```

Reply:
> ```rust
> ocaml::Error::Message
> ```

Thanks.  I tried this, but it didn't like it, so I had to wrap in `Err()` again

Reply:
> ```rust
> BigInteger256::deserialize(&mut &x[..]).map_err(|_| ocaml::Error::Message("deserialization error"))
> ```

Nice - I'll gave this form a try and it also doesn't require wrapping in the extra `Err`, so I think this is better.